bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/api/WebexTeamsApi.java

```diff
@@ -128,30 +123,29 @@ private <I, O> O doRequest(URI url, HttpMethod method, String authToken, Class<O
                 req.header("Content-type", "application/json");
             }
 
-            InputStreamResponseListener listener = new InputStreamResponseListener();
-            req.send(listener);
+            ContentResponse response = req.send();
 
-            // Wait for the response headers to arrive
-            Response response = listener.get(5, TimeUnit.SECONDS);
             logger.debug("Response: {} - {}", response.getStatus(), response.getReason());
 
             if (response.getStatus() == HttpStatus.UNAUTHORIZED_401) {
+                response.abort(new Exception(response.getReason()));
                 throw new WebexAuthenticationException();
             } else if (response.getStatus() == HttpStatus.OK_200) {
                 // Obtain the input stream on the response content
-                try (InputStream input = listener.getInputStream()) {
+                try (InputStream input = new ByteArrayInputStream(response.getContent())) {
                     Reader reader = new InputStreamReader(input);
                     O entity = gson.fromJson(reader, clazz);
                     return entity;
-                } catch (IOException e) {
-                    throw new WebexTeamsApiException("ioexception", e);
+                } catch (IOException | JsonIOException | JsonSyntaxException e) {
+                    throw new WebexTeamsApiException("exception", e);
                 }
             } else {
                 logger.warn("Unexpected response {} - {}", response.getStatus(), response.getReason());
-                try (InputStream input = listener.getInputStream()) {
+                try (InputStream input = new ByteArrayInputStream(response.getContent())) {
                     String text = new BufferedReader(new InputStreamReader(input, StandardCharsets.UTF_8)).lines()
                             .collect(Collectors.joining("\n"));
                     logger.warn("Content: {}", text);
+                    response.abort(new Exception(response.getReason()));
```

Comment:
No more needed now that you are using a synchronous HTTP call.

Reply:
Removed.